src/pks/transport/transport_ats_pk.cc

```diff
@@ -650,61 +605,28 @@ void
 Transport_ATS::InitializeFields_()
 {
   Teuchos::OSTab tab = vo_->getOSTab();
-
-  InitializeFieldFromField_(tcc_matrix_key_, tag_next_, tcc_key_, tag_next_, false, false);
   S_->GetW<CompositeVector>(solid_residue_mass_key_, tag_next_, name_).PutScalar(0.0);
   S_->GetRecordW(solid_residue_mass_key_, tag_next_, name_).set_initialized();
+  S_->GetW<CompositeVector>(conserve_qty_key_, tag_next_, name_).PutScalar(0.0);
   S_->GetRecordW(conserve_qty_key_, tag_next_, name_).set_initialized();
 }
 
-/* ****************************************************************
-* Auxiliary initialization technique.
-**************************************************************** */
-void
-Transport_ATS::InitializeFieldFromField_(const Key& field0,
-                                         const Tag& tag0,
-                                         const Key& field1,
-                                         const Tag& tag1,
-                                         bool call_evaluator,
-                                         bool overwrite)
-{
-  if (S_->HasRecord(field0, tag0)) {
-    if (S_->GetRecord(field0, tag0).owner() == name_) {
-      if ((!S_->GetRecord(field0, tag0).initialized()) || overwrite) {
-        if (call_evaluator) S_->GetEvaluator(field1, tag0).Update(*S_, name_);
-
-        const CompositeVector& f1 = S_->Get<CompositeVector>(field1, tag1);
-        CompositeVector& f0 = S_->GetW<CompositeVector>(field0, tag0, name_);
-        f0 = f1;
-
-        S_->GetRecordW(field0, tag0, name_).set_initialized();
-        if (vo_->os_OK(Teuchos::VERB_MEDIUM) && (!overwrite)) {
-          *vo_->os() << "initiliazed " << field0 << " to " << field1 << std::endl;
-        }
-      }
-    }
-  }
-}
-
 
 /* *******************************************************************
 * Estimation of the time step based on T.Barth (Lecture Notes
 * presented at VKI Lecture Series 1994-05, Theorem 4.2.2.
 * Routine must be called every time we update a flow field.
 *
 * Warning: Barth calculates influx, we calculate outflux. The methods
-* are equivalent for divergence-free flows and gurantee EMP. Outflux
+* are equivalent for divergence-free flows and guarantee EMP. Outflux
 * takes into account sinks and sources but preserves only positivity
 * of an advected mass.
 * ***************************************************************** */
 double
 Transport_ATS::StableTimeStep()
 {
   S_->Get<CompositeVector>(flux_key_, Tags::NEXT).ScatterMasterToGhosted("face");
-  flux_ = S_->Get<CompositeVector>(flux_key_, Tags::NEXT).ViewComponent("face", true);
-
-  Teuchos::RCP<Epetra_Map> cell_map =
-    Teuchos::rcp(new Epetra_Map(mesh_->getMap(AmanziMesh::Entity_kind::CELL, false)));
+  auto flux = S_->Get<CompositeVector>(flux_key_, Tags::NEXT).ViewComponent("face", true);
   IdentifyUpwindCells();
 
   tcc = S_->GetPtrW<CompositeVector>(tcc_key_, tag_current_, passwd_);
```

Comment:
Using local variables like above slightly changes the results and make some multi-scale tests failed. This needs to be verified.
```suggestion
    auto tcc_ = S_->GetPtrW<CompositeVector>(tcc_key_, tag_current_, passwd_);
    Epetra_MultiVector& tcc_prev = *tcc_->ViewComponent("cell");
```